app/uk/gov/hmrc/gform/pdf/model/PDFPageModelBuilder.scala

```diff
@@ -0,0 +1,157 @@
+        val addToListSummary: AddToListSummary =
+          AddToListSummary(
+            addToList.repeaters.last.title.value(),
+            addToList.repeaters.map(_.expandedDescription.value()).toList
+          )
+        val addToListPageGroups: List[AddToListPageGroup] = addToList.iterations.toList.zipWithIndex.flatMap {
+          case (iteration, index) =>
+            val singletonsSorted =
+              singletonWithNumberOrdering.fold(iteration.singletons.toList)(iteration.singletons.toList.sorted(_))
+            val addToListPages: List[PageData] = singletonsSorted
+              .flatMap { case SingletonWithNumber(singleton, sectionNumber) =>
+                buildFromSingleton(
+                  cache,
+                  envelopeWithMapping,
+                  singleton,
+                  sectionNumber,
+                  addToList.source,
+                  validationResult
+                )
+              }
+            val addToListIterationTitle = iteration.repeater.repeater.expandedShortName.value()
+            if (addToListPages.isEmpty) {
+              None
+            } else {
+              Some(
+                AddToListPageGroup(
+                  addToListIterationTitle,
+                  addToListPages,
+                  addToList.source.addAnotherQuestion.id.value + index
+                )
+              )
+            }
+        }
+
+        if (addToListPageGroups.isEmpty)
+          List.empty
+        else
+          List(
+            AddToListData(
+              addToListTitle,
+              addToListSummary,
+              addToListPageGroups,
+              addToList.source.addAnotherQuestion.id.value
+            )
+          )
+      }
+    }
+  }
+
+  private def buildFromSingleton[A <: PageMode, T <: PDFType](
+    cache: AuthCacheWithForm,
+    envelopeWithMapping: EnvelopeWithMapping,
+    singleton: Singleton[A],
+    sectionNumber: SectionNumber,
+    source: Section,
+    validationResult: ValidationResult
+  )(implicit
+    messages: Messages,
+    l: LangADT,
+    lise: SmartStringEvaluator,
+    pdfFunctions: PDFCustomRender[T]
+  ): Option[PageData] = {
+
+    import pdfFunctions._
+
+    val filteredFields = doFilter(singleton.page.fields)
+    val pageFields: List[PageField] = formComponentOrdering
+      .fold(filteredFields)(filteredFields.sorted(_))
+      .map(fc => PDFPageFieldBuilder.build(fc, cache, sectionNumber, validationResult, envelopeWithMapping))
+    if (pageFields.isEmpty) {
+      None
+    } else {
+      val maybePresentationHint = source.fold { _ =>
+        singleton.page.presentationHint
+      } { _ =>
+        singleton.page.presentationHint
+      } { addToList =>
+        addToList.presentationHint
+          .orElse(singleton.page.presentationHint)
+      }
+      val pageTitle = getPageTitle(singleton.page, maybePresentationHint)
+      Option(PageData(pageTitle, pageFields, sectionNumber.value.toString))
```

Comment:
Where this habit is coming from. In this particular case param to `Option.apply` will never be `null` so why don't use `Some` instead?

Reply:
My lack of knowledge on the diff between Option(x) and Some(x).